### PR TITLE
Autocomplete should keep upper-/lowercase from the word being autocompleted

### LINKED ISSUE #2181: [Innspill til komponent]: <Combobox />
### Tilbakemelding

Overskriv formattering ved valg av predefinerte verdier når man autocompleter med enter.
[Eksempel](https://github.com/navikt/aksel/assets/32221678/5b7db7ea-ed6a-49cd-a5c9-9f8f95036fef)

### PR DESCRIPTION
**Pull request by it-vegard**
### Description

Fixes #2181 

### Change summary

- Combobox autocomplete now changes upper-/lowercase to match the autocompleted word as the user types

### CODE REVIEW

File: @navikt/core/react/src/form/combobox/Input/Input.tsx [hunk 1]
```diff
@@ -46,28 +46,40 @@ const Input = forwardRef<HTMLInputElement, InputProps>(
 
     const onEnter = useCallback(
       (event: React.KeyboardEvent) => {
+        const isTextInSelectedOptions = (text: string) => {
+          return selectedOptions.find(
+            (item) => item.toLowerCase() === text.toLowerCase()
```
Comment (it-vegard) [inline @navikt/core/react/src/form/combobox/Input/Input.tsx]: Change to toLocaleLowerCase to account for differences in some languages, like Turkish:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLocaleLowerCase#description

File: @navikt/core/react/src/form/combobox/Input/Input.tsx [hunk 2]
```diff
@@ -46,28 +46,40 @@ const Input = forwardRef<HTMLInputElement, InputProps>(
...
+        } else if (shouldAutocomplete && isTextInSelectedOptions(value)) {
           event.preventDefault();
           // Trying to set the same value that is already set, so just clearing the input
           clearInput(event);
         } else if ((allowNewValues || shouldAutocomplete) && value !== "") {
           event.preventDefault();
           // Autocompleting or adding a new value
-          toggleOption(value, event);
-          if (!isMultiSelect && !selectedOptions.includes(value))
+          const selectedValue = filteredOptions[0] || value;
```
Comment (it-vegard) [inline @navikt/core/react/src/form/combobox/Input/Input.tsx]: Endre til følgende kode (husk import av isValueNew):
`const selectedValue = allowNewValues && !isValueNew(value) ? value : filteredOptions[0];`

File: @navikt/core/react/src/form/combobox/combobox.stories.tsx [hunk 3]
```diff
@@ -467,6 +467,50 @@ export const TestThatCallbacksOnlyFireWhenExpected: StoryObj<{
...
+    expect(input.value).toBe("cAmEl CaSe");
+    await userEvent.type(input, "{Enter}");
+    await sleep(250);
+    const chips = canvas.getAllByRole("list")[0];
+    const selectedUpperCaseChip = within(chips).getAllByRole("listitem")[0];
+    expect(selectedUpperCaseChip).toHaveTextContent("Camel Case"); // A weird issue is preventing the accessible name from being used in the test, even if it works in VoiceOver
+
+    // With custom option
+    userEvent.click(input);
+    await userEvent.type(input, "PaScAl CaSe", { delay: 250 });
```
Comment (it-vegard) [inline @navikt/core/react/src/form/combobox/combobox.stories.tsx]: Endre til "cAmEl" (partial match)

File: .changeset/early-colts-sneeze.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@navikt/ds-react": patch
+---
+
+Autocomplete in combobox will now change formatting of the typed word to match upper-/lowercase of word being autocompleted
```
Comment (it-vegard) [inline .changeset/early-colts-sneeze.md]: Change this to say we DO NOT change formatting during typing, but use casing from the autocompleted word when selecting/pressing enter.

**Review by HalvorHaugan — APPROVED** [review]

**Pull request merged**